### PR TITLE
Feat/#8

### PR DESCRIPTION
**Pull request by DungNguyen2003**
refer feat/#8

### CODE REVIEW

File: src/app.ts [hunk 1]
```diff
@@ -50,6 +52,13 @@ initializeMiddlewares();
 app.use(APP_CONSTANTS.apiPrefix, routers);
 initializeErrorHandler();
 
+app.get('/login',(req, res) => {
+  res.json({
+    data : login
+  })
+})
+
+
```
Comment (nc-minh) [inline src/app.ts:61]: move to controller not here

File: src/middlewares/auth.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    }
+    catch (err) {
+        new ApiResponse(err,'Login failed', 401).send(res);
+
+    }
+};
+
+const adminMiddleware = (req : Request, res : Response, next: NextFunction) =>{
+    const user = req.body.user;
+    if(!user || user.roles != 'ADMIN'){
```
Comment (nc-minh) [inline src/middlewares/auth.ts:25]: Roles is enum, use enum instead of string 'ADMIN'

File: src/utils/rest/response.ts [hunk 3]
```diff
@@ -1,9 +1,28 @@
 /**
  * Standard API response object
  */
-export default interface ApiResponse {
-  data?: any;
-  meta?: { [key: string]: any };
-  errors?: Error;
-  message?: string;
+import {Response} from 'express'
```
Comment (nc-minh) [inline src/utils/rest/response.ts:4]: revert

File: src/utils/formatter.ts [hunk 4]
```diff
@@ -32,16 +33,20 @@ export class Formatter {
       data = result;
     }
```
Comment (nc-minh) [inline src/utils/formatter.ts:35]: revert

**Review by nc-minh — CHANGES_REQUESTED** [review]
login not object, you have to move objects related to auth into auth folder

File: src/apis/v1/login/service.ts [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+async function checkRole(email: string){
+    const user = await UserModel.findOne({ email});
+
+    return user ? user.roles : null;
+}
+
+const login = async function (req: Request) {
+    const tokenGoogle = req.headers.token_google;
+    
+    const EXPIRESIN = process.env.EXPIRESIN || '10d'
```
[the remark this answers is not in the document]
Comment (DungNguyen2003) [reply src/apis/v1/login/service.ts:38]: e để trong env cái biến EXPIRESIN là '7d', nếu mà k có thì là 10 ngày

**Pull request merged**